nixosModules/homestakeros/options.nix

```diff
@@ -1,6 +1,19 @@
 { lib
+, cfg
 , ...
-}: {
+}:
+let
+  # Try to get the first mount's path, fallback to null
```

Comment:
Why default to null when you match on it later on to make it a string anyway?
Null is an anti-pattern: your return "type" is basically `Either` ( `String` | *Undefined* ).
Null is very bad because it has so much undefined behavior when its being ran against library functions, especially in languages like Nix which do not have strong typing.